### PR TITLE
Update DockerBuilder tags input data handling.

### PR DESCRIPTION
**Pull request by KostyaSha**
Space separated tags -> new line separated.

### CODE REVIEW

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java [hunk 1]
```diff
@@ -39,20 +45,64 @@
...
+    public void setTagsString(String tagsString) {
+        setTags(filterStringToList(tagsString));
+    }
+
+    public static List<String> filterStringToList(String str) {
+        return str == null ? null : Splitter.on("\n").omitEmptyStrings().trimResults().splitToList(str);
+    }
+
+    public static void verifyTags(String tagsString) {
+        final Pattern pattern = Pattern.compile("^([a-z0-9-_.]+)$");
```
Comment (lanwen) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java]: please move pattern to const

**Comment by lanwen** [comment]
what exactly does this PR?

**Comment by KostyaSha** [comment]
Changes space separated values into new line separated stored in list.

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java (hunk 1 above)
Comment (KostyaSha) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java]: done

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java [hunk 2]
```diff
@@ -25,34 +21,88 @@
...
+    }
+
+    public static List<String> filterStringToList(String str) {
+        return str == null ? null : Splitter.on("\n").omitEmptyStrings().trimResults().splitToList(str);
+    }
+
+    public static void verifyTags(String tagsString) {
+        final List<String> verifyTags = filterStringToList(tagsString);
+        for (String verifyTag : verifyTags) {
+            if (!VALID_REPO_PATTERN.matcher(verifyTag).matches()) {
```
Comment (lanwen) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java]: can avoid if with one of lang3.Validate methods.

File: docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java [hunk 3]
```diff
@@ -25,34 +21,88 @@
...
+                throw new IllegalArgumentException("Tag " + verifyTag + " doesn't match ^([a-z0-9-_.]+)$");
```
Comment (lanwen) [inline docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java]: can you use const here? Or instead you can change const in future, but can forget to change it here.

**Comment by lanwen** [comment]
ok, :+1:. I'll refactor this builder in future

**Comment by KostyaSha** [comment]
Tested locally

**Pull request merged**